### ISSUE TITLE
Enabling/Disabling of service fails if enable_cmd is insserv and insserv returns warnings

### ISSUE THREAD

**Issue by TheSameCH** [ansible/ansible-modules-core#1502]
This example message lets the task fail although rc of insserv is 0:

```
"msg": 
   "Failed to install service. rc: 0, out: , 
    err: insserv: warning: script 'S17lockgcstartup' missing LSB tags and overrides\n
    insserv: warning: script 'K01gcstartup' missing LSB tags and overrides\n 
    insserv: warning: script 'K01unlockgcstartup' missing LSB tags and overrides\n
    insserv: warning: script 'lockgcstartup' missing LSB tags and overrides\n
    insserv: warning: script 'unlockgcstartup' missing LSB tags and overrides\n
    insserv: warning: current stop runlevel(s) (3 5) of script `unlockgcstartup' overwrites defaults (empty).\n
    insserv: Default-Start undefined, assuming default start runlevel(s) for script `unlockgcstartup'\n
    insserv: script nsrps is broken: incomplete LSB comment.\n
    insserv: missing `Provides:' entry: please add.\n
    insserv: warning: script 'init.ohasd' missing LSB tags and overrides\n
    insserv: Default-Start undefined, assuming default start runlevel(s) for script `init.ohasd'\n
    insserv: warning: script 'gcstartup' missing LSB tags and overrides\n
    insserv: Service network is missed in the runlevels 4 to use service rhnsd\n
    insserv: Service syslog is missed in the runlevels 4 to use service aurad\n",
"failed": true
```

OS: SLES 11 SP2/3

**Comment by jctanner** [comment]
Reproduced on SLES11sp3 vagrant box with the following init script ...

```
#!/bin/sh
#
#     Template SUSE system startup script for example service/daemon FOO
#     Copyright (C) 1995--2005  Kurt Garloff, SUSE / Novell Inc.
#          
#     This library is free software; you can redistribute it and/or modify it
#     under the terms of the GNU Lesser General Public License as published by
#     the Free Software Foundation; either version 2.1 of the License, or (at
#     your option) any later version.
#                 
#     This library is distributed in the hope that it will be useful, but
#     WITHOUT ANY WARRANTY; without even the implied warranty of
#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
#     Lesser General Public License for more details.
#      
#     You should have received a copy of the GNU Lesser General Public
#     License along with this library; if not, write to the Free Software
#     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
#     USA.
#
# /etc/init.d/FOO
#   and its symbolic link
# /(usr/)sbin/rcFOO
#
# Template system startup script for some example service/daemon FOO
#
# LSB compatible service control script; see http://www.linuxbase.org/spec/
# 
# Note: This template uses functions rc_XXX defined in /etc/rc.status on
# UnitedLinux/SUSE/Novell based Linux distributions. If you want to base your
# script on this template and ensure that it works on non UL based LSB 
# compliant Linux distributions, you either have to provide the rc.status
# functions from UL or change the script to work without them.
# See skeleton.compat for a template that works with other distros as well.
#
### BEGIN INIT INFO
# Provides:          FOOBAR
# Required-Start:    $syslog $remote_fs
# Should-Start:      $time ypbind smtp
# Required-Stop:     $syslog $remote_fs
# Should-Stop:       
# Default-Stop:      0 1 2 6
# Short-Description: FOO XYZ daemon providing ZYX
# Description:       Start FOO to allow XY and provide YZ
#   continued on second line by '#<TAB>'
#   should contain enough info for the runlevel editor
#   to give admin some idea what this service does and
#   what it's needed for ...
#   (The Short-Description should already be a good hint.)
### END INIT INFO
# 
# Any extensions to the keywords given above should be preceeded by 
# X-VendorTag- (X-UnitedLinux- X-SuSE- for us) according to LSB.
# 
# Notes on Required-Start/Should-Start:
# * There are two different issues that are solved by Required-Start
#    and Should-Start
# (a) Hard dependencies: This is used by the runlevel editor to determine
#     which services absolutely need to be started to make the start of
#     this service make sense. Example: nfsserver should have
#     Required-Start: $portmap
#     Also, required services are started before the dependent ones.
#     The runlevel editor will warn about such missing hard dependencies
#     and suggest enabling. During system startup, you may expect an error,
#     if the dependency is not fulfilled.
# (b) Specifying the init script ordering, not real (hard) dependencies.
#     This is needed by insserv to determine which service should be
#     started first (and at a later stage what services can be started
#     in parallel). The tag Should-Start: is used for this.
#     It tells, that if a service is available, it should be started
#     before. If not, never mind.
# * When specifying hard dependencies or ordering requirements, you can 
#   use names of services (contents of their Provides: section)
#   or pseudo names starting with a $. The following ones are available
#   according to LSB (1.1):
#   $local_fs       all local file systems are mounted
#               (most services should need this!)
#   $remote_fs      all remote file systems are mounted
#               (note that /usr may be remote, so
#                many services should Require this!)
#   $syslog         system logging facility up
#   $network        low level networking (eth card, ...)
#   $named          hostname resolution available
#   $netdaemons     all network daemons are running
#   The $netdaemons pseudo service has been removed in LSB 1.2.
#   For now, we still offer it for backward compatibility.
#   These are new (LSB 1.2):
#   $time           the system time has been set correctly  
#   $portmap        SunRPC portmapping service available
#   UnitedLinux extensions:
#   $ALL            indicates that a script should be inserted
#               at the end
# * The services specified in the stop tags 
#   (Required-Stop/Should-Stop)
#   specify which services need to be still running when this service
#   is shut down. Often the entries there are just copies or a subset 
#   from the respective start tag.
# * Should-Start/Stop are now part of LSB as of 2.0,
#   formerly SUSE/Unitedlinux used X-UnitedLinux-Should-Start/-Stop.
#   insserv does support both variants.
# * X-UnitedLinux-Default-Enabled: yes/no is used at installation time
#   (%fillup_and_insserv macro in %post of many RPMs) to specify whether
#   a startup script should default to be enabled after installation.
#   It's not used by insserv.
#
# Note on runlevels:
# 0 - halt/poweroff             6 - reboot
# 1 - single user           2 - multiuser without network exported
# 3 - multiuser w/ network (text mode)  5 - multiuser w/ network and X11 (xdm)
# 
# Note on script names:
# http://www.linuxbase.org/spec/refspecs/LSB_1.3.0/gLSB/gLSB/scrptnames.html
# A registry has been set up to manage the init script namespace.
# http://www.lanana.org/
# Please use the names already registered or register one or use a
# vendor prefix.


# Check for missing binaries (stale symlinks should not happen)
# Note: Special treatment of stop for LSB conformance

# Source LSB init functions
# providing start_daemon, killproc, pidofproc, 
# log_success_msg, log_failure_msg and log_warning_msg.
# This is currently not used by UnitedLinux based distributions and
# not needed for init scripts for UnitedLinux only. If it is used,
# the functions from rc.status should not be sourced or used.
#. /lib/lsb/init-functions

# Shell functions sourced from /etc/rc.status:
#      rc_check         check and set local and overall rc status
#      rc_status        check and set local and overall rc status
#      rc_status -v     be verbose in local rc status and clear it afterwards
#      rc_status -v -r  ditto and clear both the local and overall rc status
#      rc_status -s     display "skipped" and exit with status 3
#      rc_status -u     display "unused" and exit with status 3
#      rc_failed        set local and overall rc status to failed
#      rc_failed <num>  set local and overall rc status to <num>
#      rc_reset         clear both the local and overall rc status
#      rc_exit          exit appropriate to overall rc status
#      rc_active        checks whether a service is activated by symlinks
. /lib/lsb/init-functions
. /etc/rc.status

# Reset status of this service
rc_reset

# Return values acc. to LSB for all commands but status:
# 0   - success
# 1       - generic or unspecified error
# 2       - invalid or excess argument(s)
# 3       - unimplemented feature (e.g. "reload")
# 4       - user had insufficient privileges
# 5       - program is not installed
# 6       - program is not configured
# 7       - program is not running
# 8--199  - reserved (8--99 LSB, 100--149 distrib, 150--199 appl)
# 
# Note that starting an already running service, stopping
# or restarting a not-running service as well as the restart
# with force-reload (in case signaling is not supported) are
# considered a success.

case "$1" in
    start)
    echo -n "Starting foobar "
    ## Start daemon with startproc(8). If this fails
    ## the return value is set appropriately by startproc.
    touch /tmp/foobar.running

    log_warning_msg "THIS IS A WARNING!!!"

    # Remember status and be verbose
    rc_status -v
    ;;
    stop)
    echo -n "Shutting down foobar "
    ## Stop daemon with killproc(8) and if this fails
    ## killproc sets the return value according to LSB.
    rm -f /tmp/foobar.running

    # Remember status and be verbose
    rc_status -v
    ;;
    restart)
    ## Stop the service and regardless of whether it was
    ## running or not, start it again.
    $0 stop
    $0 start

    ;;
    status)
    echo -n "Checking for service foobar "
    ## Check status with checkproc(8), if process is running
    ## checkproc will return with exit status 0.

    # Return value is slightly different for the status command:
    # 0 - service up and running
    # 1 - service dead, but /var/run/  pid  file exists
    # 2 - service dead, but /var/lock/ lock file exists
    # 3 - service not running (unused)
    # 4 - service status unknown :-(
    # 5--199 reserved (5--99 LSB, 100--149 distro, 150--199 appl.)

    # NOTE: checkproc returns LSB compliant status values.
    # NOTE: rc_status knows that we called this init script with
    # "status" option and adapts its messages accordingly.
    #rc_status -v
    if [ -f /tmp/foobar.running ]; then
        echo -n "running"
        echo ""
        #exit 0
    else
        echo -n "dead"
        echo ""
        exit 3
    fi
    ;;
    *)
    echo "Usage: $0 {start|stop|status|restart}"
    exit 1
    ;;
esac
rc_exit
```

The key element to trigger the warning was to remove the "Default-Start" line, which then throws warnings with the insserv command ...

```
sles11-sp3:/etc/init.d # insserv -r foobar
insserv: script foobar is broken: incomplete LSB comment.
insserv: missing `Default-Start:'  entry: please add even if empty.
insserv: script foobar is broken: incomplete LSB comment.
insserv: missing `Default-Start:'  entry: please add even if empty.
insserv: script foobar is broken: incomplete LSB comment.
insserv: missing `Default-Start:'  entry: please add even if empty.
insserv: script foobar is broken: incomplete LSB comment.
insserv: missing `Default-Start:'  entry: please add even if empty.
insserv: script foobar is broken: incomplete LSB comment.
insserv: missing `Default-Start:'  entry: please add even if empty.
insserv: script foobar is broken: incomplete LSB comment.
insserv: missing `Default-Start:'  entry: please add even if empty.
insserv: warning: current stop runlevel(s) (3 5) of script `foobar' overwrites defaults (empty).
insserv: Default-Start undefined, assuming default start runlevel(s) for script `foobar'
insserv: script foobar.good: service FOOBAR already provided!
```

Service module error ...

```
TASK [service] *****************************************************************
task path: /workspace/issues/AMC-1502/site.yml:5
<10.0.0.113> ESTABLISH SSH CONNECTION FOR USER: root
<10.0.0.113> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-wi
th-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/jtanner/.ansible/cp/ansible
-ssh-%h-%p-%r 10.0.0.113 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1462411591.77-242126523920353 `" && echo "` ech
o $HOME/.ansible/tmp/ansible-tmp-1462411591.77-242126523920353 `" )'"'"''
<10.0.0.113> PUT /tmp/tmp3aGMOA TO /root/.ansible/tmp/ansible-tmp-1462411591.77-242126523920353/service
<10.0.0.113> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gss
api-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/jtanner/.ansible/cp/a
nsible-ssh-%h-%p-%r '[10.0.0.113]'
<10.0.0.113> ESTABLISH SSH CONNECTION FOR USER: root
<10.0.0.113> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-wi
th-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/jtanner/.ansible/cp/ansible
-ssh-%h-%p-%r -tt 10.0.0.113 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tm
p-1462411591.77-242126523920353/service; rm -rf "/root/.ansible/tmp/ansible-tmp-1462411591.77-242126523920353/" > /dev/null 2>&1'"'"''
fatal: [slesbox]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"arguments": "", "enabled": true, "name": "foobar", "pat
tern": null, "runlevel": "default", "sleep": null, "state": null}, "module_name": "service"}, "msg": "Failed to install service. rc: 0, out: , err: insserv: script foobar is broken: incomplete LSB comment.\ninsserv: missing `Default-Start:'  entry: please add even if empty.\ninsserv: script foobar is broken: incomplete LSB comment.\ninsserv: missing `Default-Start:'  entry: please add even if empty.\ninsserv: Default-Start undefined, assuming defau$t start runlevel(s) for script `foobar'\ninsserv: script foobar.good: service FOOBAR already provided!\n"}
```

**Comment by jctanner** [comment]
Resolved by merging https://github.com/ansible/ansible-modules-core/pull/1846

**Comment by jctanner** [comment]
Sorry, we changed our mind and think that the behavior should remain the same. If the script is "broken", as noted by the errors from insserv, they should be resolved instead of ignored.
